drivers/sensor/lps22hb/lps22hb.c

```diff
@@ -0,0 +1,161 @@
+/* lps22hb.c - Driver for LPS22HB pressure and temperature sensor */
+
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sensor.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <misc/byteorder.h>
+#include <misc/__assert.h>
+
+#include "lps22hb.h"
+
+static inline int lps22hb_set_odr_raw(struct device *dev, u8_t odr)
+{
+	struct lps22hb_data *data = dev->driver_data;
+	const struct lps22hb_config *config = dev->config->config_info;
+
+	return i2c_reg_update_byte(data->i2c_master, config->i2c_slave_addr,
+				   LPS22HB_REG_CTRL_REG1,
+				   LPS22HB_MASK_CTRL_REG1_ODR,
+				   odr << LPS22HB_SHIFT_CTRL_REG1_ODR);
+}
+
+static int lps22hb_sample_fetch(struct device *dev,
+				enum sensor_channel chan)
+{
+	struct lps22hb_data *data = dev->driver_data;
+	const struct lps22hb_config *config = dev->config->config_info;
+	u8_t out[5];
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	if (i2c_burst_read(data->i2c_master, config->i2c_slave_addr,
+			   LPS22HB_REG_PRESS_OUT_XL, (u8_t *)out, 5) < 0) {
+		SYS_LOG_DBG("Failed to read sample");
+		return -EIO;
+	}
+
+	data->sample_press = (s32_t)((u32_t)(out[0]) |
+				     ((u32_t)(out[1]) << 8) |
+				     ((u32_t)(out[2]) << 16));
+	data->sample_temp = (s16_t)((u16_t)(out[3]) |
+				    ((u16_t)(out[4]) << 8));
+
+	return 0;
+}
+
+static inline void lps22hb_press_convert(struct sensor_value *val,
+					 s32_t raw_val)
+{
+	/* val = raw_val / 4096 */
```

Comment:
Please expand on the comment to clarify where the 4096 comes from. Something like, "Raw pressure data has a resolution of 4096 LSB/hPa." I also think you may be off by an order of magnitude, since 1 hPa=0.1 kPa, and the sensor interface defines pressure in kPa.